src/annotator/anchoring/test/types-test.js

```diff
@@ -19,43 +21,290 @@ describe('types', function () {
+      });
+
+      it('returns a RangeSelector if options are missing', () => {
+        const range = new RangeAnchor(container, createRangeAnchor());
+        assert.deepEqual(range.toSelector(), rangeSelectorResult());
+      });
+    });
+  });
+
+  describe('TextPositionAnchor', () => {
+    let fakePosFromRange;
+    let fakePosToRange;
+
+    beforeEach(() => {
+      fakePosFromRange = sinon.stub();
+      fakePosToRange = sinon.stub();
+      $imports.$mock({
+        'dom-anchor-text-position': {
+          fromRange: fakePosFromRange,
+          toRange: fakePosToRange,
+        },
+      });
+    });
+
+    afterEach(() => {
+      $imports.$restore();
+    });
+
+    function createTextPositionAnchor() {
+      return new TextPositionAnchor(container, 0, 0);
+    }
+
+    describe('#fromRange', () => {
+      it('returns a TextPositionAnchor instance', () => {
+        fakePosFromRange.returns({
+          start: 0,
+          end: 0,
+        });
+        const anchor = TextPositionAnchor.fromRange(container, new Range());
+        assert.calledOnce(fakePosFromRange);
+        assert.instanceOf(anchor, TextPositionAnchor);
+      });
+    });
+
+    describe('#fromSelector', () => {
+      it('returns a TextPositionAnchor instance', () => {
+        const anchor = TextPositionAnchor.fromSelector(container, {
+          start: 0,
+          end: 0,
+        });
+        assert.instanceOf(anchor, TextPositionAnchor);
+      });
+    });
+
+    describe('#toSelector', () => {
+      it('returns a TextPositionSelector', () => {
+        const anchor = createTextPositionAnchor();
+        assert.deepEqual(anchor.toSelector(), {
+          type: 'TextPositionSelector',
+          start: 0,
+          end: 0,
+        });
+      });
+    });
+
+    describe('#toRange', () => {
+      it('returns a range object', () => {
+        const anchor = createTextPositionAnchor();
+        fakePosToRange.returns('fake range');
+        assert.equal(anchor.toRange(), 'fake range');
+        assert.calledWith(fakePosToRange, container, { start: 0, end: 0 });
+      });
+    });
+  });
+
+  describe('TextQuoteAnchor', () => {
+    let fakeQuoteToRange;
+    let fakeQuoteFromRange;
+    let fakeToTextPosition;
+
+    beforeEach(() => {
+      fakeQuoteToRange = sinon.stub();
+      fakeQuoteFromRange = sinon.stub();
+      fakeToTextPosition = sinon.stub();
+      $imports.$mock({
+        'dom-anchor-text-quote': {
+          fromRange: fakeQuoteFromRange,
+          toRange: fakeQuoteToRange,
+          toTextPosition: fakeToTextPosition,
+        },
+      });
+    });
+
+    afterEach(() => {
+      $imports.$restore();
+    });
+
+    function createTextQuoteAnchor() {
+      return new TextQuoteAnchor(container, 'Liberty', {
+        prefix: 'a new nation, conceived in ',
+        suffix: ', and dedicated to the proposition that',
+      });
+    }
+
+    it('creates a TextQuoteAnchor if `context` is missing', () => {
```

Comment:
If you replace the `context` fallback with a default parameter value then this test will be less necessary.